File: cpp/350_Intersection_of_Two_Arrays_II.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
public:
	vector<int> intersect(vector<int>& n1, vector<int>& n2) {
		vector<int> freq(1001, 0);

		for (int n : n1) freq[n]++;

		vector<int> ret;
		for (int n : n2) {
			if (freq[n] > 0) {
				ret.push_back(n);
				--freq[n];
			}
		}
		return ret;
	}
};

File: cpp/350_Intersection_of_Two_Arrays_II.py
class Solution:
    def intersect(self, n1: List[int], n2: List[int]) -> List[int]:
        freq = [0] * 1001

        for n in n1:
            freq[n] += 1

        ret = []

        for n in n2:
            if freq[n] > 0:
                ret.append(n)
                freq[n] -= 1

        return ret

